instrumentation/spring/spring-boot-autoconfigure/build.gradle.kts

```diff
@@ -3,6 +3,8 @@ plugins {
   id("otel.publish-conventions")
 }
 
+// Name the Spring Boot modules in accordance with https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.developing-auto-configuration.custom-starter
+base.archivesName.set("opentelemetry-spring-boot")
```

Comment:
can we rename the directory instead?

Reply:
We could -- I just thought that a directory named just `spring-boot` would be a tad more confusing than `spring-boot-autoconfigure` when you look at what the `spring` dir contains